chrome/content/zotero/xpcom/zotero.js

```diff
@@ -507,6 +509,35 @@ Components.utils.import("resource://gre/modules/osfile.jsm");
 			}
 		}
 		
+		// Check for DB purge
+		var newAccountFile = dataDir.clone();
+		newAccountFile.append('new-account');
```

Comment:
When you unlink, it asks if you want to delete your local data (defaulting to no, probably), and if you do it kicks off the same process (but with no username, so forced auto-sync should be a no-op).